src/io/s3_filesys.cc

```diff
@@ -64,70 +66,289 @@ struct XMLIter {
     return true;
   }
 };
+
+/*!
+ * \brief Converts hash to hex representation
+ * \param hash unsigned char array with hash
+ * \param size size of hash
+ * \return string in hex representation
+ */
+static const std::string SHA256HashToHex(unsigned char *hash, int size) {
+  CHECK_EQ(size, SHA256_DIGEST_LENGTH);
+  std::stringstream ss;
+  for (int i=0; i < SHA256_DIGEST_LENGTH; i++) {
+    ss << std::hex << std::setw(2) << std::setfill('0') << static_cast<int>(hash[i]);
```

Comment:
should you save current formatting modes and restore?

Reply:
I couldn't find a reference on if that is possible. Besides this is a static method...